src/main/webapp/app/scores/exercise-scores.component.html

```diff
@@ -198,6 +205,37 @@ <h2>
                         ></jhi-programming-assessment-manual-result>
                     </ng-template>
                 </ngx-datatable-column>
+
+                <ngx-datatable-column prop="" *ngIf="exercise.type === ExerciseType.FILE_UPLOAD" [resizeable]="false" [minWidth]="180" [width]="180">
+                    <ng-template ngx-datatable-cell-template let-value="value">
+                        <button routerLink="/file-upload-exercise/{{ exercise.id }}/submission/{{ value.submission.id }}/assessment" class="btn btn-primary btn-sm mr-1">
+                            <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>
+                            {{ 'artemisApp.fileUploadSubmission.view' | translate }}
+                        </button>
+                    </ng-template>
+                </ngx-datatable-column>
+
+                <ngx-datatable-column prop="" *ngIf="exercise.type === ExerciseType.MODELING" [resizeable]="false" [minWidth]="180" [width]="180">
+                    <ng-template ngx-datatable-cell-template let-value="value">
+                        <button
+                            routerLink="/modeling-exercise/{{ exercise.id }}/submissions/{{ value.submission.id }}/assessment"
+                            [queryParams]="{ optimal: false, hideBackButton: true }"
+                            class="btn btn-primary btn-sm mr-1"
+                        >
+                            <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>
+                            {{ 'artemisApp.modelingSubmission.view' | translate }}
+                        </button>
+                    </ng-template>
+                </ngx-datatable-column>
+
+                <ngx-datatable-column prop="" *ngIf="exercise.type === ExerciseType.TEXT" [resizeable]="false" [minWidth]="180" [width]="180">
+                    <ng-template ngx-datatable-cell-template let-value="value">
+                        <button routerLink="/text/{{ exercise.id }}/assessment/{{ value.submission.id }}" class="btn btn-primary btn-sm mr-1">
```

Comment:
Why are you navigating to the assessment page here, but for the other 2 exercise types, you navigate to the submission page?

I think this should be consistent for all exercise types! 

We also need to make sure that tutors can not navigate to arbitrary assessments on this page because this might be problematic for our double blind assessment mechanism

Reply:
As discussed, all of them lead to the assessment page of the submission for now (in the future they might lead to a read only version of only the submission) and the buttons will only be shown to instructors of that exercise.

Reply:
I just tested the instructor only display of the new "View submission" buttons on the test server but maybe someone else can verify it as well.